packages/gafl-webapp-service/src/handlers/agreed-handler.js

```diff
@@ -168,7 +169,7 @@ const processPayment = async (request, transaction, status) => {
       status[COMPLETION_STATUS.paymentFailed] = true
       status.payment = { code: state.code }
       await request.cache().helpers.status.set(status)
-      next = PAYMENT_FAILED.uri
+      next = addLanguageCodeToUri(request, PAYMENT_FAILED.uri)
```

Comment:
I think that this and the other failures need more specific tests. Let's work through this together when you're in next week